network/linux_s/linux_s.h

```diff
@@ -0,0 +1,101 @@
+ * patent holders to use this software.
+ * - Use of the software either in source or binary form, must be run
+ * on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+#ifndef NOOS_NETWORK_LINUX_S_LINUX_S_H_
+#define NOOS_NETWORK_LINUX_S_LINUX_S_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "network_interface.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define SOCKET_NO 4000
+#define SOCKETS (SOCKET_NO + 1)
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+struct lserver_desc {
+	/* Server fd */
+	int32_t	fd;
+	/* Server port */
+	uint16_t	port;
+	/* Ids of clientes socket that are available*/
+	uint32_t	client_ids[SOCKET_NO];
+	/* Number of cliente socket available*/
+	uint32_t	back_log_clients;
+};
+
+
+struct linux_desc {
+	/*Id of the open socket*/
+	uint32_t id;
+
+	/*Sockets*/
+	int32_t sockets[SOCKETS];
+
+	/*Server descriptor*/
+	struct lserver_desc server;
+
+	/*Reference to the network*/
+	struct network_interface net;
+};
+
+struct linux_init_param {
+	/*	Maximum size for incoming data
+		uint32_t max_buff_size;*/
+
+	/* Reference to the network*/
+	struct network_interface* net;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+int32_t linux_get_network_interface(struct linux_desc *desc,
+				    struct network_interface** net);
+
```

Comment:
if you remove linux_get_network_interface, in this file you will just need:
extern struct network_interface linux_net;